docs/components/rule.md

```diff
@@ -1,42 +1,104 @@
 
-The data of each rule node can be labeled to satisfy the clusters labeling scheme. High-availability pairs can be run in parallel and should be distinguished by the designated replica label, just like regular Prometheus servers.
+You can think of Rule as a simplified Prometheus that does not require a sidecar and does not collect and do PromQL evaluation (no QueryAPI).
+
+The data of each Rule node can be labeled to satisfy the clusters labeling scheme. High-availability pairs can be run in parallel and should be distinguished by the designated replica label, just like regular Prometheus servers.
+Read more about Ruler in HA in [here](rule.md#Ruler_HA)
 
 ```
 $ thanos rule \
-    --data-dir          "/path/to/data" \
-    --eval-interval     "30s" \
-    --rule-file         "/path/to/rules/*.rules.yaml" \
-    --alert.query-url   "http://0.0.0.0:9090" \
-    --alertmanagers.url "alert.thanos.io" \
-    --cluster.peers     "thanos-cluster.example.org" \
-    --objstore.config-file "bucket.yml"
+    --data-dir             "/path/to/data" \
+    --eval-interval        "30s" \
+    --rule-file            "/path/to/rules/*.rules.yaml" \
+    --alert.query-url      "http://0.0.0.0:9090" \ # This tells what query URL to link to in UI.
+    --alertmanagers.url    "alert.thanos.io" \
+    --query                "query.example.org" \
+    --query                "query2.example.org" \
+    --objstore.config-file "bucket.yml" \
+    --label                'monitor_cluster="cluster1"'
+    --label                'replica="A"
 ```
 
-The content of `bucket.yml`:
+## Risk
 
-```yaml
-type: GCS
-config:
-  bucket: example-bucket
-```
+Ruler has conceptual tradeoffs that might not be favorable for most use cases. The main tradeoff is its dependence on 
+query reliability. For Prometheus it is unlikely to have alert/recording rule evaluation failure as evaluation is local.
+
+For Ruler the read path is distributed, since most likely ruler is querying Thanos Querier which gets data from remote Store APIs. 
+
+This means that **query failure** are more likely to happen, that's why clear strategy on what will happen to alert and during query
+unavailability is the key.
+
+Note that Ruler will never trigger alert on it's own, so *no alert will be triggered if the query fails* without extra (mandatory)
+meta monitoring. This is exactly the same Alertmanager project. Even though there might be internal error Alertmanager will only produce metric, will 
+never notify receiver, even though it could.
+
+To be sure that alerting works it is essential to monitor Ruler and alert from another **Scraper (Prometheus + sidecar)** that sits in same cluster.
+
+You should alert on:
+
+// TODO(bwplotka) TO BE EXTENDED.
+
+ * `thanos_alert_sender_alerts_dropped_total` to check if our connection to alertmanager works.
+ * `prometheus_rule_group_last_duration_seconds < prometheus_rule_group_interval_seconds` 
```

Comment:
I'd add `prometheus_rule_evaluation_failures_total` right away that's pretty essential alert also.

Reply:
need to work on this and ensure that ruler has all of this